x-pack/plugins/endpoint/server/routes/resolver/utils/pagination.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SearchResponse } from 'elasticsearch';
+import { ResolverEvent } from '../../../../common/types';
+import { extractEventID } from './normalize';
+import { JsonObject } from '../../../../../../../src/plugins/kibana_utils/public';
+
+export interface PaginationParams {
+  size: number;
+  timestamp?: Date;
+  eventID?: string;
+}
+
+interface PaginationCursor {
+  timestamp: Date;
+  eventID: string;
+}
+
+function urlEncodeCursor(data: PaginationCursor) {
+  const value = JSON.stringify(data);
+  return Buffer.from(value, 'utf8')
+    .toString('base64')
+    .replace(/\+/g, '-')
+    .replace(/\//g, '_')
+    .replace(/=+$/g, '');
+}
+
+function urlDecodeCursor(value: string): PaginationCursor {
+  value = value.replace(/\-/g, '+').replace(/_/g, '/');
+  const data = Buffer.from(value, 'base64').toString('utf8');
```

Comment:
❔ Looks like a job for https://nodejs.org/dist/latest-v12.x/docs/api/string_decoder.html